exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java

```diff
@@ -98,6 +102,32 @@ private void throwIfClosed() throws AlreadyClosedSqlException,
     }
   }
 
+  /**
+   * Throws SqlTimeoutException if the statement has already timed out
+   * @throws AlreadyClosedSqlException   if Connection is closed
+   * @throws SqlTimeoutException         if Statement is already timed out
+   */
+  private void throwIfTimedOut() throws AlreadyClosedSqlException,
+                                        SqlTimeoutException,
+                                        SQLException {
+    if ( (statement instanceof DrillStatementImpl && ((DrillStatementImpl) statement).isTimedOut()) ||
```

Comment:
Good idea.  The original DrillStatement and DrillPreparedStatement are derived from the JDBC interfaces directly. I'll add the isTimedOut to DrillStatement and tie up the loose ends.